qfieldsync/gui/cloud_transfer_dialog.py

```diff
@@ -227,12 +227,13 @@ def show_project_compatibility_page(self):
         if feedback and feedback.count > 0:
             has_errors = len(feedback.error_feedbacks) > 0
 
-            feedback_table = CheckerFeedbackTable(feedback)
-            self.feedbackTableWrapperLayout.addWidget(feedback_table)
-            self.stackedWidget.setCurrentWidget(self.projectCompatibilityPage)
-            self.buttonBox.button(QDialogButtonBox.Apply).setVisible(True)
-            self.buttonBox.button(QDialogButtonBox.Apply).setEnabled(not has_errors)
-            self.buttonBox.button(QDialogButtonBox.Apply).setText(self.tr("Next"))
+            if self.feedbackTableWrapperLayout.count() == 0:
```

Comment:
I do not follow when `feedbackTableWrapperLayout.count()` is (not) 0? And therefore I don't know if we do (not) need an `else`?

Reply:
The showEvent can be triggered more than once when showing a dialog, hence why we had this bad issue with the feedback widget being duplicated. 

The check here insures we only add it once

Reply:
Great! Can we add a comment mentioning this?

Reply:
I guess we can also move the `has_errors` in the `if` block too

Reply:
@suricactus , we can move the whole function behind that condition.